### PR TITLE
fix: leave etcd on scale down

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: controllers/taloscontrolplane_controller.go [hunk 1]
```diff
@@ -281,18 +287,128 @@ func (r *TalosControlPlaneReconciler) scaleDownControlPlane(ctx context.Context,
 
 	oldest := machines[0]
 	for _, machine := range machines {
+		if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+			// Machine has already been deleted.
```
Comment (andrewrynhard) [inline controllers/taloscontrolplane_controller.go:291]: Redundant comment.

File: controllers/taloscontrolplane_controller.go [hunk 2]
```diff
@@ -281,18 +287,128 @@ func (r *TalosControlPlaneReconciler) scaleDownControlPlane(ctx context.Context,
...
 
-	r.Log.Info("Deleting control plane machine", "machine", oldest.Name)
+	if oldest.Status.NodeRef == nil {
+		return ctrl.Result{}, fmt.Errorf("%q machine does not have a nodeRef", oldest.Name)
+	}
+
+	newest := machines[0]
+	for _, machine := range machines {
+		if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+			// Machine has already been deleted.
```
Comment (andrewrynhard) [inline controllers/taloscontrolplane_controller.go:307]: Redundant comment.

**Review by smira — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]

**Pull request merged**